jwql/website/apps/jwql/static/js/jwql.js

```diff
@@ -931,54 +1196,71 @@ function update_sort_options(data, base_url) {
-};
+}
 
 /**
  * Updates the thumbnail-array div with interactive images of thumbnails
  * @param {Object} data - The data returned by the update_thumbnails_per_observation_page/update_thumbnails_query_page AJAX methods
  */
 function update_thumbnail_array(data) {
 
-      // Add content to the thumbail array div
+    // Add content to the thumbnail array div
     for (var i = 0; i < Object.keys(data.file_data).length; i++) {
         
         // Parse out useful variables
-        rootname = Object.keys(data.file_data)[i];
-        file = data.file_data[rootname];
-        viewed = file.viewed;
-        exp_type = file.exp_type;
-        filename_dict = file.filename_dict;
+        var rootname = Object.keys(data.file_data)[i];
+        var file = data.file_data[rootname];
+        var viewed = file.viewed;
+        var exp_type = file.exp_type;
+        var filename_dict = file.filename_dict;
 
         // Build div content
-        if (data.inst!="all") {
-            content = '<div class="thumbnail" instrument = ' + data.inst + ' detector="' + filename_dict.detector + '" proposal="' + filename_dict.program_id + '" file_root="' + rootname + '", exp_start="' + file.expstart + '" look="' + viewed + '" exp_type="' + exp_type + '">';
-            content += '<a href="/' + data.inst + '/' + rootname + '/">';
+        var instrument;
+        if (data.inst != "all") {
+            instrument = data.inst;
         } else {
-            content = '<div class="thumbnail" instrument = ' +filename_dict.instrument + ' detector="' + filename_dict.detector + '" proposal="' + filename_dict.program_id + '" file_root="' + rootname + '", exp_start="' + file.expstart + '" look="' + viewed + '" exp_type="' + exp_type + '">';
-            content += '<a href="/' + filename_dict.instrument + '/' + rootname + '/">';
+            instrument = filename_dict.instrument;
         }
-        content += '<span class="helper"></span><img id="thumbnail' + i + '" onerror="image_error(this);">';
+        var content = '<div class="thumbnail" data-instrument="' + instrument +
+                      '" data-detector="' + filename_dict.detector + '" data-proposal="' + filename_dict.program_id +
+                      '" data-file_root="' + rootname + '" data-group_root="' + filename_dict.group_root +
+                      '" data-exp_start="' + file.expstart + '" data-look="' + viewed + '" data-exp_type="' + exp_type + '">';
+        content += '<div class="thumbnail-group">'
+        content += '<a class="thumbnail-link" href="#" data-image-href="/' +
+                   instrument + '/' + rootname + '/" data-group-href="/' +
+                   instrument + '/exposure/' + filename_dict.group_root +  '">';
+        content += '<span class="helper"></span>'
+
+        // Make sure thumbnail img always has a src and alt
+        content += '<img id="thumbnail' + i +
+                   '" src="/static/img/default_thumb.png" ' +
```

Comment:
I think we use the imagenotfound instead of the plain blue.

```suggestion
                   "/static/img/imagenotfound.png";
```

Reply:
For this one -- I added the plain blue image for thumbnails that weren't found.  Prior to this change, the img tag had no src or alt added if there was an error, so it appeared as a plain white square indistinguishable from background.  I was hoping the blue square would be both more visible than no image at all, and more legible on mouseover than adding the imagenotfound.png.

The example on test had an issue with permissions on the new image, for some reason, so it wasn't showing up.  It is there now (https://jwql-test.stsci.edu/NIRCam/archive/2057/obs003/).  Can you please let me know if you still think I should swap it back to the other not found image?

Reply:
Mouseover is certainly more accessible with the blue than with "imagenotfound" (so I think I support this change).  
I'd defer to @bhilbert4 as the PO.

Once I click on a group or file, there is no longer a placeholder for a missing preview_image (I think this situation should use imagenotfound).

Reply:
There is still a default not found image on the image pages, e.g.
https://jwql-test.stsci.edu/NIRCam/jw02057003001_02104_00004_nrcb1/

The exposure page handles it differently -- it just hides viewers for images it can't find, whether because there is no such FITS file or because the preview wasn't found.  I can look into adding a fallback image instead if the file exists but there is no preview image.

Reply:
Yeah, the text in the mouseover is certainly more legible with the blue compared to white. 

I think keeping the Image Not Found fallback preview image makes sense. Some people do tend to jump onto JWQL shortly after their data are taken, but before preview images have been made. It seems like they'd tend to freak out less if they can see a message that there's no preview image yet.

Reply:
I pushed a modification to address this.  There is now a fallback div/image for each detector in the exposure that is shown on load.  If any real preview images get shown for the detector, the fallback is hidden and doesn't come back.

I think this should allow show/hide for FITS files that don't exist (like s3d for mirimage, i2d for mirifushort/long), but should show default images and links in the case where there are no previews available yet.

The change is on test. Let me know what you think.

Reply:
That looks good! And the filenames are available in the drop down as well as when hovering over the fallback preview image, so users have a way of seeing which detectors were used. And I don't see any option for mirimage s3d, like you said.

Reply:
This looks good in chrome, but a little wonky in safari.  Not sure if safari requires more deliberate sizing of the imgnotfound preview images.
https://jwql-test.stsci.edu/NIRCam/exposure/jw02057003001_02102_00001/

That said, it looks great in chrome, so if its not an incredibly easy fix I wouldn't worry about it for now, especially since the user can still tell they're temporary placeholders.  I'd rather get this all merged in.

Reply:
The layout in safari looks good, with the LW image centered below both columns of the SW. Everything's just a little stretched. 😆  In Firefox everything looks similar to how it does in Chrome.

Reply:
I'm a-ok with not addressing the safari stretch!

Reply:
Sorry, I noticed that at the same time you did -- the Safari stretch is fixed.

Reply:
I just pushed one more fix, to keep the view header/explore image links available in the case where there are no preview images yet.  I think that should be it.

Reply:
Just checked it out on test. looks great!